DetectorDescription/DDCMS/plugins/DDHGCalWafer.cc

```diff
@@ -0,0 +1,95 @@
+#include "DD4hep/DetFactoryHelper.h"
+#include "DataFormats/Math/interface/GeantUnits.h"
+#include "DetectorDescription/DDCMS/interface/DDPlugins.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#define EDM_ML_DEBUG
+
+using namespace geant_units::operators;
+
+static long  algorithm(dd4hep::Detector& /* description */,
+                       cms::DDParsingContext& ctxt,
+                       xml_h e,
+                       dd4hep::SensitiveDetector& /* sens */) {
+
+  cms::DDNamespace      ns(ctxt,e,true);
+  cms::DDAlgoArguments  args(ctxt, e);
+  // Header section of original DDHGCalWafer.h
+  double                   waferSize    = args.value<double>("WaferSize");
+  int                      cellType     = args.value<int>("CellType");
+  int                      nColumns     = args.value<int>("NColumns");
+  int                      nBottomY     = args.value<int>("NBottomY");
+  std::vector<std::string> childNames   = args.value<std::vector<std::string> >("ChildNames");
+  std::vector<int>         nCellsRow    = args.value<std::vector<int> >("NCellsRow");
+  std::vector<int>         angleEdges   = args.value<std::vector<int> >("AngleEdges");
+  std::vector<int>         detectorType = args.value<std::vector<int> >("DetectorType");
+  std::string              parentName   = args.parentName(); 
+  dd4hep::Volume           mother       = ns.volume(parentName);
+#ifdef EDM_ML_DEBUG
+  edm::LogVerbatim("HGCalGeom") << childNames.size() << " children: " 
+				<< childNames[0] << "; " << childNames[1]
+				<< " positioned in " << nCellsRow.size()
+				<< " rows and " << nColumns 
+				<< " columns with lowest column at "
+				<< nBottomY << " in mother " << parentName
+				<< " of size " << waferSize;
+  for (unsigned int k=0; k<nCellsRow.size(); ++k)
+    edm::LogVerbatim("HGCalGeom") << "[" << k << "] Ncells " << nCellsRow[k]
+				  << " Edge rotations " << angleEdges[2*k] 
+				  << ":" << angleEdges[2*k+1]
+				  << " Type of edge cells " 
+				  << detectorType[2*k] << ":"
+				  << detectorType[2*k+1];
+#endif
+
+  static const int incAlongX = 2;
+  static const int incAlongY = 6;
+  double dx = 0.5*waferSize/nColumns;
+  double dy = 0.5*dx*tan(30._deg);
+  int    ny = nBottomY;
+  int    kount(0);
+
+  for (unsigned int ir=0; ir<nCellsRow.size(); ++ir) {
+    int    nx   = 1 - nCellsRow[ir];
+    double ypos = dy*ny;
+    for (int ic=0; ic<nCellsRow[ir]; ++ic) {
+      std::string name(childNames[0]);
```

Comment:
@bsunanda - please, use string_view to avoid copying a string.

Reply:
@ianna Can you explain why this is needed - what is the comparison of cost in terms of CPU and memory

Reply:
I think ns.volume(std::string&) is the relevant method. How string_view can be used rather than string there?